packages/synapse-react-client/src/components/SynapsePortalBanners/SynapsePortalBanners.tsx

```diff
@@ -0,0 +1,152 @@
+import { useGetQueryResultBundleWithAsyncStatus } from '@/synapse-queries'
+import { useSourceAppConfigs } from '@/utils/hooks'
+import { BUNDLE_MASK_QUERY_RESULTS } from '@/utils/SynapseConstants'
+import { ChevronRightTwoTone } from '@mui/icons-material'
+import { Box, Typography } from '@mui/material'
+import {
+  ColumnSingleValueQueryFilter,
+  ColumnSingleValueFilterOperator,
+} from '@sage-bionetworks/synapse-types'
+
+export type SynapsePortalBannersProps = {
+  entityId: string
+  dataCatalogEntityId?: string
+  sourceAppConfigTableID?: string
+}
+
+/**
+ *
+ * @param props
+ */
+export default function SynapsePortalBanners({
+  entityId,
+  dataCatalogEntityId = 'syn61609402',
+  sourceAppConfigTableID = 'syn45291362',
+}: SynapsePortalBannersProps) {
+  const dataCatalogAdditionalFilters: ColumnSingleValueQueryFilter[] = [
+    {
+      concreteType:
+        'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
+      columnName: 'id',
+      operator: ColumnSingleValueFilterOperator.EQUAL,
+      values: [entityId],
+    },
+  ]
+  const { data: dataCatalogData } = useGetQueryResultBundleWithAsyncStatus({
+    entityId: dataCatalogEntityId,
+    query: {
+      sql: `SELECT appId, link FROM ${dataCatalogEntityId}`,
+      additionalFilters: dataCatalogAdditionalFilters,
+    },
+    partMask: BUNDLE_MASK_QUERY_RESULTS,
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+  })
+
+  const rowSet = dataCatalogData?.responseBody?.queryResult?.queryResults
+  const hasPortalBanners = !!rowSet && rowSet?.rows.length > 0
+
+  const appIds = rowSet?.rows.map(row => row.values[0]) as string[]
+
+  const sourceAppConfigFilters: ColumnSingleValueQueryFilter[] = [
+    {
+      concreteType:
+        'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
+      columnName: 'appId',
+      operator: ColumnSingleValueFilterOperator.EQUAL,
+      values: appIds,
+    },
+  ]
+  const sourceAppConfigs = useSourceAppConfigs(
+    sourceAppConfigTableID,
+    sourceAppConfigFilters,
+  )
+  const hasSourceAppConfigs = !!sourceAppConfigs && sourceAppConfigs.length > 0
+
+  if (!hasPortalBanners || !hasSourceAppConfigs) {
+    return <></>
+  }
```

Comment:
Wonder if it makes sense to show a skeleton when loading. Maybe, or maybe not if most page visits are to entities not in the data catalog

Reply:
Most entities will not have portal banners